tests/lib/ocpp/v201/functional_blocks/test_data_transfer.cpp

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Pionix GmbH and Contributors to EVerest
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+#include <message_dispatcher_mock.hpp>
+#include <ocpp/v201/functional_blocks/data_transfer.hpp>
+
+using namespace ocpp::v201;
+using ::testing::_;
+using ::testing::Invoke;
+
+// Test when the data_transfer_callback is not set, expecting DataTransferStatusEnum::UnknownVendorId
+TEST(DataTransferTest, HandleDataTransferReq_NoCallback) {
+    MockMessageDispatcher mock_dispatcher;
+    DataTransfer data_transfer(mock_dispatcher, std::nullopt); // No callback
+
+    DataTransferRequest request;
+    request.vendorId = "test_vendor";
+    ocpp::Call<DataTransferRequest> call(request, "unique_id_123");
+
+    EXPECT_CALL(mock_dispatcher, dispatch_call_result(_)).WillOnce(Invoke([](const json& call_result) {
```

Comment:
I feel like this is going to be worth investing some time in making these expectations nicer. Way outside of scope for now but would be smart to do before we make more of these tests.

I don't think we really care whether `dispatch_call` or `dispatch_call_async` is called for example, as long as one of those gets called for tests where that is a thing.

Also I would like to create custom matchers to set expectations on specified fields of structs so you don't have to do the invoke construction.